plugins/net_plugin/net_plugin.cpp

```diff
@@ -1614,20 +1614,32 @@ namespace eosio {
       return true;
    }
 
-   void sync_manager::sync_reset_lib_num(const connection_ptr& c) {
+   void sync_manager::sync_reset_lib_num(const connection_ptr& c, bool closing) {
       std::unique_lock<std::mutex> g( sync_mtx );
       if( sync_state == in_sync ) {
          sync_source.reset();
       }
       if( !c ) return;
-      if( c->current() ) {
+      if( !closing && c->current() ) {
          std::lock_guard<std::mutex> g_conn( c->conn_mtx );
          if( c->last_handshake_recv.last_irreversible_block_num > sync_known_lib_num ) {
             sync_known_lib_num = c->last_handshake_recv.last_irreversible_block_num;
          }
-      } else if( c == sync_source ) {
-         sync_last_requested_num = 0;
-         request_next_chunk( std::move(g) );
+      } else if ( closing ) {
```

Comment:
can you elaborate this idea with updating `sync_known_lib_num`, please? perhaps in PR description.
From first look I don't get why we are requesting now next chunk only if connection is closing and it was a sync source.

Reply:
I'll add a comment here, I agree it is not clear.